### PR TITLE
Add sub item groups

### PR DESCRIPTION
**Pull request by ybw0014**
Added sub item groups.

Haven't tested yet, will test it later.

### CODE REVIEW

**Comment by GallowsDove** [comment]
Looks good, just add materials to the subcategories

**Comment by ybw0014** [comment]
It is ready for review now.

File: src/main/java/me/gallowsdove/foxymachines/Items.java [hunk 1]
```diff
@@ -22,14 +23,55 @@
...
+        new NamespacedKey(FoxyMachines.getInstance(), "sacrificial_altars"),
+        MAIN_ITEM_GROUP,
+        new CustomItemStack(Material.POLISHED_BLACKSTONE_BRICKS, "&4Sacrificial Altar")
+    );
+
+    public static SubItemGroup BOSSES_ITEM_GROUP = new SubItemGroup(
+        new NamespacedKey(FoxyMachines.getInstance(), "bosses"),
+        MAIN_ITEM_GROUP,
+        new CustomItemStack(Material.DRAGON_HEAD, "&cBosses")
+    );
```
Comment (GallowsDove) [inline src/main/java/me/gallowsdove/foxymachines/Items.java:66]: This item group should only be created if custom mobs are allowed in config

File: src/main/java/me/gallowsdove/foxymachines/Items.java [hunk 2]
```diff
@@ -22,14 +23,55 @@
...
+
+    public static SubItemGroup GHOST_BLOCKS_ITEM_GROUP = new SubItemGroup(
             new NamespacedKey(FoxyMachines.getInstance(), "ghost_blocks"),
-            new CustomItemStack(Material.GLASS, "&5Ghost Blocks", "", "&a> Click to open"));
+            MAIN_ITEM_GROUP,
+            new CustomItemStack(Material.GLASS, "&5Ghost Blocks")
+    );
```
Comment (GallowsDove) [inline src/main/java/me/gallowsdove/foxymachines/Items.java:72]: Same here, this shouldn't be created if ghost blocks are disabled in config

**Review by GallowsDove — CHANGES_REQUESTED** [review]

**Comment by GallowsDove** [comment]
Looks good, thanks for the contribution.

**Pull request merged**